pkgs/development/compilers/edk2/default.nix

```diff
@@ -50,13 +66,14 @@ edk2 = stdenv.mkDerivation {
     homepage = https://sourceforge.net/projects/edk2/;
     license = stdenv.lib.licenses.bsd2;
     branch = "UDK2017";
-    platforms = ["x86_64-linux" "i686-linux" "aarch64-linux"];
+    platforms = ["x86_64-linux" "i686-linux" "aarch64-linux" "armv7l-linux"];
   };
 
   passthru = {
+    inherit targetArch hostArch;
     setup = projectDscPath: attrs: {
-      buildInputs = [ pythonEnv ] ++
-        stdenv.lib.optionals (attrs ? buildInputs) attrs.buildInputs;
+      buildInputs = [] ++ attrs.buildInputs or [];
+      nativeBuildInputs = [ buildPythonEnv buildPackages.iasl ] ++ attrs.nativeBuildInputs or [];
```

Comment:
Actually, it's even more complex because the package using the edk should always live in a different bootstrapping phase (hence my suggestion above that OVMF should use `buildPackages.edk2`). This means that this `setup` really ought to be defined elsewhere so that anything used here can come from the phase of things using edk, rather than the phase providing edk. Otherwise all the splicing stuff / manual `buildPackages` / `targetPackages` will make no sense.

What I just wrote probably in general makes no sense :). But it is exactly analogous how stdenv provided tools from the previous stage and lives in the same stage as things that use it, rather than being defined in the previous stage with the tools, and being used in the next stage. `setup` is a wrapper around `stdenv.mkDerivation`, so just as stdenv use buildPackages.gcc`, so should `setup` use `buildPackages.edk2`. But then it makes little sense to put as a passthru on `edk2` since then we spuriously reference `edk` to get `setup` which just uses `buildPackages.edk` + all the splicing issues I referenced above.